src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -237,9 +237,10 @@ trait FastSync {
         bestBlockHeaderNumber)
     }
 
-    private def handleFailingMptPeers: Receive ={
-      case BlockChainOnlyDownload(peer) =>
-        blockChainOnlyPeers = (blockChainOnlyPeers + peer).take(blockChainOnlyPeersPoolSize)
+    private def handleFailingMptPeers: Receive = {
+      case MarkPeerBlockchainOnly(peer) => if (!blockChainOnlyPeers.contains(peer)) {
+        blockChainOnlyPeers = peer +: blockChainOnlyPeers.dropRight(1)
```

Comment:
If this is the only place the `blockChainOnlyPeers` seq grows then doesn't this mean it will always be size 1?